src/services/oss/backend.rs

```diff
@@ -0,0 +1,655 @@
+            .header(header::CONTENT_TYPE, "application/octet-stream")
+            .header(header::DATE, timestamp);
+
+        if offset.unwrap_or_default() != 0 || size.is_some() {
+            req = req.header(header::RANGE, BytesRange::new(offset, size).to_string());
+        }
+
+        let req = req
+            .body(AsyncBody::Empty)
+            .map_err(|e| new_request_build_error(Operation::Read, path, e))?;
+
+        Ok(req)
+    }
+
+    fn delete_object_request(&self, path: &str) -> Result<Request<AsyncBody>> {
+        let p = build_abs_path(&self.root, path);
+
+        let url = format!("{}/{}", self.endpoint, percent_encode_path(&p));
+
+        let timestamp = reqsign::time::format_rfc2822(time::OffsetDateTime::now_utc());
+        let mut req = Request::delete(&url);
+        req = req
+            .header("Host", &self.host)
+            .header(header::DATE, timestamp);
+
+        let req = req
+            .body(AsyncBody::Empty)
+            .map_err(|e| new_request_build_error(Operation::Delete, path, e))?;
+
+        Ok(req)
+    }
+
+    fn head_object_request(&self, path: &str) -> Result<Request<AsyncBody>> {
+        let p = build_abs_path(&self.root, path);
+
+        let url = format!("{}/{}", self.endpoint, percent_encode_path(&p));
+
+        let timestamp = reqsign::time::format_rfc2822(time::OffsetDateTime::now_utc());
+        let mut req = Request::head(&url);
+        req = req
+            .header(header::HOST, &self.host)
+            .header(header::DATE, timestamp);
+
+        let req = req
+            .body(AsyncBody::Empty)
+            .map_err(|e| new_request_build_error(Operation::Stat, path, e))?;
+
+        Ok(req)
+    }
+
+    fn list_object_request(&self, path: &str, token: String) -> Result<AsyncReq> {
+        let p = build_abs_path(&self.root, path);
+
+        let url = format!(
+            "{}/?list-type=2&delimiter=/&prefix={}&continuation-token={}",
+            self.endpoint,
+            percent_encode_path(&p),
+            token
+        );
+
+        let timestamp = reqsign::time::format_rfc2822(time::OffsetDateTime::now_utc());
+
+        let req = Request::get(&url)
+            .header(header::HOST, &self.host)
+            .header(header::DATE, timestamp)
+            .body(AsyncBody::Empty)
+            .map_err(|e| new_request_build_error(Operation::List, path, e))?;
+        Ok(req)
+    }
+
+    async fn get_object(
+        &self,
+        path: &str,
+        offset: Option<u64>,
+        size: Option<u64>,
+    ) -> Result<AsyncResp> {
+        let req = self.get_object_request(path, offset, size)?;
+
+        self.sign_and_send(req, Operation::Read, path).await
+    }
+
+    async fn put_object(&self, path: &str) -> Result<AsyncResp> {
+        let req = self.put_object_request(path, Some(0), AsyncBody::Empty)?;
+        self.sign_and_send(req, Operation::Create, path).await
+    }
+
+    async fn stat_object(&self, path: &str) -> Result<AsyncResp> {
```

Comment:
There is no API in OSS called `stat_object`. Please use the same name used in OSS docs.